src/resolvers/ciResolver/impl/resolvers/githubActions/githubActionsConfigParser.ts

```diff
@@ -0,0 +1,143 @@
+import { ISpecificCIResolverResponse } from '../../../interfaces/ISpecificCIResolver';
+import { injectable } from 'inversify';
+import { ILoggerFactory } from '../../../../../utils/logger';
+import { ILogger } from '../../../../../utils/logger/interfaces/ILogger';
+import { INode, keySorter, NodeSorter, objectIterator } from '../../../../../utils/objectIterator/objectIterator';
+import { is, isArrayOfObjects, isNumber, isString } from 'ts-type-guards';
+import { INvmHandler } from '../../../interfaces/INvmHandler';
+
+const envRegex = /\${{\s*(?:env|matrix|secrets)\.([^\s]+)\s*}}/i;
+const nodeEnvReplaceRegex = /\${{\s*(?:env|matrix|secrets)\.([^\s]+)\s*}}/i;
+
+type Scalar = string | number;
+type Matrix = Record<string, string[]>;
```

Comment:
It is a bit different in other revolvers and dependes on their structure. Since it is only used internally in each resolver, I don't think it needs to be global